CommandLine.c

```diff
@@ -130,10 +130,10 @@ static CommandLineSettings parseArguments(const char* program, int argc, char**
       switch (opt) {
          case 'h':
             printHelpFlag(program);
-            exit(0);
+            return 1;
```

Comment:
Avoid magic numbers. Define a small enum for proper initialization outcomes and return the appropriate constant instead.